src/peer_manager.rs

```diff
@@ -467,14 +467,15 @@ impl PeerManager {
     pub fn verified_candidate_info(
         &self,
         log_ident: &LogIdent,
-    ) -> Result<(PublicId, Authority<XorName>), RoutingError> {
-        let (new_pub_id, new_client_auth) = match self.candidate {
+    ) -> Result<(PublicId, Authority<XorName>, PublicId), RoutingError> {
```

Comment:
Should update the doc comment of this method to reflect the changed return type. I'd also consider turning that tuple into a struct with named fields to make things clearer.

Reply:
Or maybe return `(PublicId, OnlinePayload)` ?